test/unit/remote/watcher.js

```diff
@@ -0,0 +1,418 @@
+            rev: '9-8765'
+          }
+        }
+      }
+
+      await this.pull.onChange(clone(doc))
+
+      this.prep.moveDocAsync.called.should.be.true()
+      let src = this.prep.moveDocAsync.args[0][2]
+      src.should.have.properties({
+        path: 'my-folder/file-2',
+        docType: 'file',
+        checksum: doc.md5sum,
+        tags: doc.tags,
+        remote: {
+          _id: '12345678902'
+        }
+      })
+      let dst = this.prep.moveDocAsync.args[0][1]
+      dst.should.have.properties({
+        path: doc.path,
+        docType: 'file',
+        checksum: doc.md5sum,
+        tags: doc.tags,
+        remote: {
+          _id: doc._id,
+          _rev: doc._rev
+        }
+      })
+      dst.should.not.have.properties(['_rev', 'path', 'name'])
+    })
+
+    it('calls deleteDoc & addDoc when file has changed completely', async function () {
+      this.prep.deleteDocAsync = sinon.stub()
+      this.prep.addDocAsync = sinon.stub()
+      this.prep.deleteDocAsync.returnsPromise().resolves(null)
+      this.prep.addDocAsync.returnsPromise().resolves(null)
+      let doc: RemoteDoc = {
+        _id: '12345678903',
+        _rev: '6-abcdef',
+        _type: FILES_DOCTYPE,
+        type: 'file',
+        dir_id: 'whatever',
+        path: 'another-folder/in/some/place',
+        name: 'file-3-bis',
+        md5sum: '8888888888888888888888888888888888888888',
+        tags: [],
+        created_at: '2017-01-30T09:09:15.217662611+01:00',
+        updated_at: '2017-01-30T09:09:15.217662611+01:00'
+      }
+      let was: Metadata = await this.pouch.db.get('my-folder/file-3')
+      was.remote._rev = doc._rev
+      await this.pouch.db.put(was)
+
+      await this.pull.onChange(clone(doc))
+
+      this.prep.deleteDocAsync.called.should.be.true()
+      let id = this.prep.deleteDocAsync.args[0][1].path
+      id.should.equal('my-folder/file-3')
+      this.prep.addDocAsync.called.should.be.true()
+      let args = this.prep.addDocAsync.args[0]
+      args[0].should.equal('remote')
+      args[1].should.have.properties({
+        path: doc.path,
+        docType: 'file',
+        checksum: doc.md5sum,
+        tags: doc.tags,
+        remote: {
+          _id: doc._id,
+          _rev: doc._rev
+        }
+      })
+      args[1].should.not.have.properties(['_rev', 'path', 'name'])
+    })
+
+    it('calls deleteDoc for a deleted doc', async function () {
+      this.prep.deleteDocAsync = sinon.stub()
+      this.prep.deleteDocAsync.returnsPromise().resolves(null)
+      let doc = {
+        _id: '12345678901',
+        _rev: '7-abcdef',
+        _deleted: true
+      }
+
+      await this.pull.onChange(doc)
+
+      this.prep.deleteDocAsync.called.should.be.true()
+      let id = this.prep.deleteDocAsync.args[0][1].path
+      id.should.equal('my-folder/file-1')
+    })
+
+    it('calls addDoc for folder created by the mobile app', async function () {
+      // FIXME: What makes a folder created by the mobile app special?
```

Comment:
I think this is the mobile test I was thinking, it's not related to the checksum actually.
Still not sure what makes it special... the folder name?
@nono ?